apps/system/js/notifications.js

```diff
@@ -456,10 +468,31 @@ var NotificationScreen = {
     message.classList.add('detail');
     var messageContent = document.createElement('div');
     messageContent.classList.add('detail-content');
-    messageContent.textContent = detail.text;
+    // As per the UX,in text messages notification text will be name of
+    // the sender and the text of message.
+    if(detail.manifestURL === smsManifestUrl) {
+      messageContent.textContent = detail.title + ': ' + detail.text;
+    } else {
+      messageContent.textContent = detail.text;
+    }
     messageContent.setAttribute('dir', 'auto');
     message.appendChild(messageContent);
     notificationNode.appendChild(message);
+    } else {
+      // in case if notification is already present on the screen, its counter
+      // value will be incremented.
+      var notificationNode = document.querySelector('.notification');
+      var countVal = this.unreadNotifications[cIndex].count;
+      this.unreadNotifications[cIndex].count = ++countVal;
+      document.getElementById(manifestURL).querySelector('.detail-count').textContent = countVal;
```

Comment:
you are calling the same function ``document.getElementById(manifestURL)`` 3 times in this block, it will be more efficient if you call it once, assign returned value to the variable and then use that variable in all 3 places.